File: src/main/java/Characters/SyujinkouState.java
//　主人公の状態を管理する

package Characters;

public enum SyujinkouState {

	// 通常状態(速度倍率1.0)
	NORMAL(1.0),

	// フィーバー状態(速度倍率1.2にすることでEnemyより早くなり食べられる。)
	FEVER(1.2);

	// 「速度倍率」を保存しておくためのspeedMultiplier
	private final double speedMultiplier;

	// ==================================================
	// コンストラクタ
	// ==================================================
	// 速度倍率を設定する
	SyujinkouState(double speedMultiplier) {
		this.speedMultiplier = speedMultiplier;
	}

	// ==================================================
	// getter
	// ==================================================
	// 速度倍率を取得する
	public double getSpeedMultiplier() {
		return speedMultiplier;
	}

}

File: src/main/java/Characters/SyujinkouState.js
// Characters/SyujinkouState.js

/**
 * 主人公（仙石さん）の状態を管理するオブジェクト（JavaのEnumの代わり）
 * 各状態に「状態名(name)」と「速度倍率(speedMultiplier)」を持たせています。
 */
export const SyujinkouState = Object.freeze({
    
    // 通常状態 (速度倍率 1.0)
    NORMAL: Object.freeze({
        name: "NORMAL",
        speedMultiplier: 1.0
    }),

    // フィーバー状態 (速度倍率 1.2 にすることでEnemyより早くなり、食べることができる)
    FEVER: Object.freeze({
        name: "FEVER",
        speedMultiplier: 1.2
    })
});